pkg/util/serviceaccount.go

```diff
@@ -83,14 +83,26 @@ func WaitForServiceAccountSecretCreation(client kubeclient.Interface, asObj *cor
 			}
 			return false, fmt.Errorf("failed to retrieve service account(%s/%s) from cluster, err: %v", asObj.Namespace, asObj.Name, err)
 		}
-		clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return false, nil
+		clusterSecret, err = GetSecret(client, serviceAccount.Namespace, serviceAccount.Name)
+		if apierrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: serviceAccount.Namespace,
+					Name:      serviceAccount.Name,
+					Annotations: map[string]string{
+						"kubernetes.io/service-account.name": serviceAccount.Name,
```

Comment:
We can use `corev1.ServiceAccountNameKey` here.

Reply:
fixed,thanks